substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/HeapPolicyOptions.java

```diff
@@ -53,6 +53,9 @@
     @Option(help = "How many bytes is enough to allocate an unaligned chunk for an array?  0 implies (AlignedHeapChunkSize / 8).") //
     public static final HostedOptionKey<Long> LargeArrayThreshold = new HostedOptionKey<>(HeapPolicy.LARGE_ARRAY_THRESHOLD_SENTINEL_VALUE);
 
+    @Option(help = "The maximum percentage of heap free after GC to avoid shrinking") //
+    public static final RuntimeOptionKey<Integer> MaxHeapFreeRatio = new RuntimeOptionKey<>(70);
```

Comment:
`70%` sound like a very high value. Considering that the default maximum heap size is also a large fraction of the physical memory size, this means that we are basically never returning chunks back to the operating system.

What workloads did you use to evaluate the change? How does it affect performance vs. memory footprint?

Reply:
70% comes from the default of the same value in JDK. It seems a bit confusing compared to the previous logic: before we kept chunks when used < minheap, now we keep chunks when *free* < MaxHeapFree. Which means that with a value of 70%, we keep chunks when used is > 30%. Does that still sound bad?